test/metabase/notification/send_test.clj

```diff
@@ -621,11 +621,30 @@
         ; Wait for thread to complete
         (.join ^Thread thread 1000)
 
-        (is (= {:id 42 :payload_type :notification/testing :test-value "X"} @result))))))
+        (is (= {:id 42 :payload_type :notification/testing :test-value "X"} @result))))
+
+    (testing "queue blocks when full"
```

Comment:
This test seems to be inside `notification-dedup-priority-test`, which is a little confusing.

Maybe time to break up this compound test into smaller cases that can run in parallel?